lib/ansible/modules/cloud/hcloud/hcloud_ssh_key.py

```diff
@@ -0,0 +1,219 @@
+            - Only required if no ssh_key I(name) is given
+        type: int
+    name:
+        description:
+            - The Name of the Hetzner Cloud ssh_key to manage.
+            - Only required if no ssh_key I(id) is given or a ssh_key does not exists.
+        type: str
+    fingerprint:
+        description:
+            - The Fingerprint of the Hetzner Cloud ssh_key to manage.
+            - Only required if no ssh_key I(id) or I(name) is given.
+        type: str
+    labels:
+        description:
+            - User-defined labels (key-value pairs)
+        type: dict
+    public_key:
+        description:
+            - The Public Key to add.
+            - Required if ssh_key does not exists.
+        type: str
+    state:
+        description:
+            - State of the ssh_key.
+        default: present
+        choices: [ absent, present ]
+        type: str
+extends_documentation_fragment: hcloud
+"""
+
+EXAMPLES = """
+- name: Create a basic ssh_key
+  hcloud_ssh_key:
+    name: my-ssh_key
+    public_key: "ssh-rsa AAAjjk76kgf...Xt"
+    state: present
+
+- name: Ensure the ssh_key is absent (remove if needed)
+  hcloud_ssh_key:
+    name: my-ssh_key
+    state: absent
+"""
+
+RETURN = """
+hcloud_ssh_key:
+    description: The ssh_key instance
+    returned: Always
+    type: dict
+    sample: {
+        "id": 1937415,
+        "name": "mein-ssh_key",
+        "fingerprint": "b7:2f:30:a0:2f:6c:58:6c:21:04:58:61:ba:06:3b:2f",
+        "public_key": "ssh-rsa AAAjjk76kgf...Xt",
+        "labels":{}
+    }
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from ansible.module_utils.hcloud import Hcloud
+
+try:
+    from hcloud.volumes.domain import Volume
+    from hcloud.ssh_keys.domain import SSHKey
+    from hcloud.ssh_keys.domain import Server
+    from hcloud import APIException
+except ImportError:
+    pass
+
+
+class AnsibleHcloudSSHKey(Hcloud):
+    def __init__(self, module):
+        Hcloud.__init__(self, module, "hcloud_ssh_key")
+        self.hcloud_ssh_key = None
+
+    def _prepare_result(self):
+        return {
+            "id": to_native(self.hcloud_ssh_key.id),
+            "name": to_native(self.hcloud_ssh_key.name),
+            "fingerprint": to_native(self.hcloud_ssh_key.fingerprint),
+            "public_key": to_native(self.hcloud_ssh_key.public_key),
+            "labels": self.hcloud_ssh_key.labels,
+        }
+
+    def _get_ssh_key(self):
+        try:
+            if self.module.params.get("id") is not None:
+                self.hcloud_ssh_key = self.client.ssh_keys.get_by_id(
+                    self.module.params.get("id")
+                )
+            elif self.module.params.get("name") is not None:
+                self.hcloud_ssh_key = self.client.ssh_keys.get_by_name(
+                    self.module.params.get("name")
+                )
+            elif self.module.params.get("fingerprint") is not None:
```

Comment:
would it make sense to change the order to select by fingerprint before the name? Would this allow to update a ssh key's name by giving an id or fingerprint?